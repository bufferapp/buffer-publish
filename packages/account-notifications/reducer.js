import { actionTypes as dataFetchActionTypes } from '@bufferapp/async-data-fetch';
import keyWrapper from '@bufferapp/keywrapper';

export const actionTypes = keyWrapper('EMAIL_PREFERENCES', {
  TOGGLE_NOTIFICATIONS: 0,
});

export const initialState = {
  collaboration: null,
  queue: null,
  newsletter: null,
  milestones: null,
  postFailure: null,
};

export default (state = initialState, action) => {
  switch (action.type) {
    case `user_${dataFetchActionTypes.FETCH_SUCCESS}`:
      return {
        ...state,
        ...action.result.hasEmailNotifications,
      };
    case `setNotifications_${dataFetchActionTypes.FETCH_START}`:
      return {
        ...state,
        ...action.args.notifications,
      };
    case `setNotifications_${dataFetchActionTypes.FETCH_FAIL}`: {
      const rollbackState = {};
      Object.keys(action.args.notifications).map(type => {
        rollbackState[type] = !action.args.notifications[type];
        return null;
      });
      return {
        ...state,
        ...rollbackState,
      };
    }
    default:
      return state;
  }
};

export const actions = {
  handleToggleClick: notifications => ({
    type: actionTypes.TOGGLE_NOTIFICATIONS,
    notifications,
  }),
};
