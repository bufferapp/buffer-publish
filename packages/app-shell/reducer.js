import keyWrapper from '@bufferapp/keywrapper';
import { actionTypes as dataFetchActionTypes } from '@bufferapp/async-data-fetch';

export const actionTypes = keyWrapper('APP_SHELL', {
  SET_BANNER_OPTIONS: 0,
  ON_CLOSE_BANNER: 0,
});

export const initialState = {
  showReturnToClassic: false,
  showUpgradeToPro: false,
  user: {
    name: '...',
    email: '',
    avatar: null,
  },
  bannerKey: null,
  bannerOptions: undefined,
};

export default (state = initialState, action) => {
  switch (action.type) {
    case `user_${dataFetchActionTypes.FETCH_SUCCESS}`:
      return {
        ...state,
        user: {
          email: action.result.email,
          name: action.result.name,
        },
        showReturnToClassic: action.result.showReturnToClassic,
        showUpgradeToPro: action.result.is_free_user,
        showManageTeam: !action.result.is_free_user,
        showStartProTrial: action.result.canStartProTrial,
      };
    case actionTypes.SET_BANNER_OPTIONS:
      return {
        ...state,
        bannerKey: action.key,
        bannerOptions: {
          text: action.text,
          actionButton: action.actionButton,
          customHTML: action.customHTML,
          themeColor: action.themeColor,
        },
      };
    default:
      return state;
  }
};

export const actions = {
  setBannerOptions: ({ key, text, actionButton, customHTML, themeColor }) => ({
    type: actionTypes.SET_BANNER_OPTIONS,
    key,
    text,
    actionButton,
    customHTML,
    themeColor,
  }),
  onCloseBanner: ({ key }) => ({
    type: actionTypes.ON_CLOSE_BANNER,
    key,
  }),
};
