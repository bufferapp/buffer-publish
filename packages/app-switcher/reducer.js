import {
  actionTypes as dataFetchActionTypes,
  actions as dataFetchActions,
} from '@bufferapp/async-data-fetch';
import keyWrapper from '@bufferapp/keywrapper';

export const actionTypes = keyWrapper('APPSWITCHER', {
  SHOW_FEEDBACK_MODAL: 0,
  HIDE_FEEDBACK_MODAL: 0,
});

export const initialState = {
  redirecting: false,
  showGoBackToClassic: false,
  submittingFeedback: false,
  showFeedbackModal: false,
  user: {
    loading: true,
  },
};

export default (state = initialState, action) => {
  switch (action.type) {
    case `user_${dataFetchActionTypes.FETCH_SUCCESS}`:
      return {
        ...state,
        // User is not in phased rollout - so they can switch
        showGoBackToClassic: !action.result.hasNewPublish,
        user: {
          ...action.result,
          loading: false,
        },
      };
    case `sendFeedback_${dataFetchActionTypes.FETCH_START}`:
      return {
        ...state,
        submittingFeedback: true,
      };
    case `sendFeedback_${dataFetchActionTypes.FETCH_SUCCESS}`:
      return {
        ...state,
        redirecting: true,
      };
    case `sendFeedback_${dataFetchActionTypes.FETCH_FAIL}`:
      return {
        ...state,
        submittingFeedback: false,
      };
    case actionTypes.SHOW_FEEDBACK_MODAL:
      return {
        ...state,
        showFeedbackModal: true,
        source: action.source,
      };
    case actionTypes.HIDE_FEEDBACK_MODAL:
      return {
        ...state,
        showFeedbackModal: false,
      };
    default:
      return state;
  }
};

export const actions = {
  closeFeedbackModal: ({ source }) => ({
    type: actionTypes.HIDE_FEEDBACK_MODAL,
    source,
  }),
  displayFeedbackModal: ({ source }) => ({
    type: actionTypes.SHOW_FEEDBACK_MODAL,
    source,
  }),
  sendFeedback: ({ feedback, source }) =>
    dataFetchActions.fetch({ name: 'sendFeedback', args: { body: feedback, source } }),
};
