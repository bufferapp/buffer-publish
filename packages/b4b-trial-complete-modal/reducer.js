import keyWrapper from '@bufferapp/keywrapper';

export const actionTypes = keyWrapper('B4B_TRIAL_COMPLETE_MODAL', {
  CANCEL_TRIAL: 0,
});

export const initialState = {
};

export default (state = initialState, action) => {
  switch (action.type) {
    default:
      return state;
  }
};

export const actions = {
  cancelTrial: () => ({
    type: actionTypes.CANCEL_TRIAL,
  }),
};
