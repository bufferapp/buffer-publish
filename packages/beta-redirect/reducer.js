import {
  actionTypes as dataFetchActionTypes,
} from '@bufferapp/async-data-fetch';

export const actionTypes = {};

const initialState = {
  hasPublishBeta: false,
  hasPublishBetaRedirect: false,
  hasNewPublish: false,
  loading: true,
  onPaydayPage: false,
  isOnAwesomePlan: false,
};

export default (state = initialState, action) => {
  switch (action.type) {
    case `user_${dataFetchActionTypes.FETCH_SUCCESS}`: {
      const { result: { hasNewPublish, isOnAwesomePlan, features = [] } } = action;
      return {
        loading: false,
        hasPublishBeta: features.includes('new_publish_beta'),
        hasPublishBetaRedirect: features.includes('new_publish_beta_redirect'),
        hasNewPublish,
        onPaydayPage: window.location.pathname.endsWith('plans'),
        isOnAwesomePlan,
      };
    }
    default:
      return state;
  }
};

export const actions = {};
