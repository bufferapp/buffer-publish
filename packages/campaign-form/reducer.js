import keyWrapper from '@bufferapp/keywrapper';
import { actionTypes as dataFetchActionTypes } from '@bufferapp/async-data-fetch';

export const actionTypes = keyWrapper('CAMPAIGNS', {
  CREATE_CAMPAIGN: 0,
  EDIT_CAMPAIGN: 0,
});

export const initialState = {
  campaign: {},
  isLoading: false,
  campaignId: null,
};

export default (state = initialState, action) => {
  switch (action.type) {
    case `createCampaign_${dataFetchActionTypes.FETCH_START}`: {
      return {
        ...state,
        isLoading: true,
      };
    }

    case `createCampaign_${dataFetchActionTypes.FETCH_SUCCESS}`:
    case `getCampaign_${dataFetchActionTypes.FETCH_SUCCESS}`: {
      return {
        ...state,
        campaign: action.result,
        campaignId: action.result.id,
        isLoading: false,
      };
    }

    case `createCampaign_${dataFetchActionTypes.FETCH_FAIL}`:
      return {
        ...state,
        isLoading: false,
      };
    default:
      return state;
  }
};

export const actions = {
  handleCreateCampaignClick: ({ name, color }) => ({
    type: actionTypes.CREATE_CAMPAIGN,
    name,
    color,
  }),
  handleEditCampaignClick: ({ id, name, color, orgId }) => ({
    type: actionTypes.EDIT_CAMPAIGN,
    id,
    name,
    color,
    orgId,
  }),
};
