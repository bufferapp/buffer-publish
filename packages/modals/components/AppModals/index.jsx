import React from 'react';
import PropTypes from 'prop-types';

import UpgradeModal from '@bufferapp/publish-upgrade-modal';
import WelcomeModal from '@bufferapp/publish-welcome-modal';
import WelcomePaidModal from '@bufferapp/publish-welcome-paid-modal';
import ProfilesDisconnectedModal from '@bufferapp/publish-profiles-disconnected-modal';

const AppModals = ({ showUpgradeModal, showWelcomeModal, showWelcomePaidModal, showProfilesDisconnectedModal }) => (
  <React.Fragment>
    {showUpgradeModal && <UpgradeModal />}
    {showWelcomeModal && <WelcomeModal />}
    {showWelcomePaidModal && <WelcomePaidModal />}
    {showProfilesDisconnectedModal && <ProfilesDisconnectedModal />}
  </React.Fragment>
);

AppModals.propTypes = {
  showUpgradeModal: PropTypes.bool.isRequired,
  showWelcomeModal: PropTypes.bool.isRequired,
  showWelcomePaidModal: PropTypes.bool.isRequired,
  showProfilesDisconnectedModal: PropTypes.bool.isRequired,
};

export default AppModals;
