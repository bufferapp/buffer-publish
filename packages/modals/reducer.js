import keyWrapper from '@bufferapp/keywrapper';

export const initialState = {
  showUpgradeModal: false,
  showWelcomeModal: false,
  showWelcomePaidModal: false,
  showProfilesDisconnectedModal: false,
  upgradeModalSource: null,
  showStealProfileModal: false,
  stealProfileUsername: null,
  showInstagramDirectPostingModal: false,
  showWelcomeB4BTrialModal: false,
  showInstagramFirstCommentModal: false,
  showInstagramNewFirstCommentUserModal: false,
  showB4BTrialExpiredModal: false,
  upgradeModalB4BSource: null,
};

export const actionTypes = keyWrapper('MODALS', {
  SHOW_UPGRADE_MODAL: 0,
  HIDE_UPGRADE_MODAL: 0,
  SHOW_WELCOME_MODAL: 0,
  HIDE_WELCOME_MODAL: 0,
  SHOW_WELCOME_PAID_MODAL: 0,
  HIDE_WELCOME_PAID_MODAL: 0,
  SHOW_PROFILES_DISCONNECTED_MODAL: 0,
  HIDE_PROFILES_DISCONNECTED_MODAL: 0,
  SHOW_STEAL_PROFILE_MODAL: 0,
  HIDE_STEAL_PROFILE_MODAL: 0,
  SHOW_IG_DIRECT_POSTING_MODAL: 0,
  HIDE_IG_DIRECT_POSTING_MODAL: 0,
  SHOW_WELCOME_B4B_TRIAL_MODAL: 0,
  HIDE_WELCOME_B4B_TRIAL_MODAL: 0,
  SHOW_UPGRADE_B4B_MODAL: 0,
  HIDE_UPGRADE_B4B_MODAL: 0,
  SHOW_INSTAGRAM_FIRST_COMMENT_MODAL: 0,
  HIDE_INSTAGRAM_FIRST_COMMENT_MODAL: 0,
  SHOW_INSTAGRAM_NEW_FIRST_COMMENT_USER_MODAL: 0,
  HIDE_INSTAGRAM_NEW_FIRST_COMMENT_USER_MODAL: 0,
  SHOW_INSTAGRAM_FIRST_COMMENT_START_TRIAL_MODAL: 0,
  HIDE_INSTAGRAM_FIRST_COMMENT_START_TRIAL_MODAL: 0,
});

export default (state = initialState, action) => {
  switch (action.type) {
    case actionTypes.SHOW_UPGRADE_MODAL:
      return {
        ...state,
        showUpgradeModal: true,
        upgradeModalSource: action.source,
      };
    case actionTypes.HIDE_UPGRADE_MODAL:
      return {
        ...state,
        showUpgradeModal: false,
      };
    case actionTypes.SHOW_UPGRADE_B4B_MODAL:
      return {
        ...state,
        showB4BTrialExpiredModal: true,
        upgradeModalB4BSource: action.source,
      };
    case actionTypes.HIDE_UPGRADE_B4B_MODAL:
      return {
        ...state,
        showB4BTrialExpiredModal: false,
      };
    case actionTypes.SHOW_WELCOME_MODAL:
      return {
        ...state,
        showWelcomeModal: true,
      };
    case actionTypes.SHOW_INSTAGRAM_FIRST_COMMENT_MODAL:
      return {
        ...state,
        showInstagramFirstCommentModal: true,
        firstCommentIds: action.ids,
      };
    case actionTypes.SHOW_INSTAGRAM_NEW_FIRST_COMMENT_USER_MODAL:
      return {
        ...state,
        showInstagramNewFirstCommentUserModal: true,
      };
    case actionTypes.HIDE_WELCOME_MODAL:
      return {
        ...state,
        showWelcomeModal: false,
      };
    case actionTypes.SHOW_WELCOME_PAID_MODAL:
      return {
        ...state,
        showWelcomePaidModal: true,
      };
    case actionTypes.HIDE_WELCOME_PAID_MODAL:
      return {
        ...state,
        showWelcomePaidModal: false,
      };
    case actionTypes.HIDE_INSTAGRAM_FIRST_COMMENT_MODAL:
      return {
        ...state,
        showInstagramFirstCommentModal: false,
        firstCommentIds: null,
      };
    case actionTypes.HIDE_INSTAGRAM_NEW_FIRST_COMMENT_USER_MODAL:
      return {
        ...state,
        showInstagramNewFirstCommentUserModal: false,
      };
    case actionTypes.SHOW_PROFILES_DISCONNECTED_MODAL:
      return {
        ...state,
        showProfilesDisconnectedModal: true,
      };
    case actionTypes.HIDE_PROFILES_DISCONNECTED_MODAL:
      return {
        ...state,
        showProfilesDisconnectedModal: false,
      };
    case actionTypes.SHOW_STEAL_PROFILE_MODAL:
      return {
        ...state,
        showStealProfileModal: true,
        stealProfileUsername: action.stealProfileUsername,
      };
    case actionTypes.HIDE_STEAL_PROFILE_MODAL:
      return {
        ...state,
        showStealProfileModal: false,
      };
    case actionTypes.SHOW_IG_DIRECT_POSTING_MODAL:
      return {
        ...state,
        showInstagramDirectPostingModal: true,
      };
    case actionTypes.HIDE_IG_DIRECT_POSTING_MODAL:
      return {
        ...state,
        showInstagramDirectPostingModal: false,
      };
    case actionTypes.SHOW_WELCOME_B4B_TRIAL_MODAL:
      return {
        ...state,
        showWelcomeB4BTrialModal: true,
      };
    case actionTypes.HIDE_WELCOME_B4B_TRIAL_MODAL:
      return {
        ...state,
        showWelcomeB4BTrialModal: false,
      };
    case actionTypes.HIDE_INSTAGRAM_FIRST_COMMENT_START_TRIAL_MODAL:
      return {
        ...state,
        showInstagramFirstCommentStartTrialModal: false,
      };
    case actionTypes.SHOW_INSTAGRAM_FIRST_COMMENT_START_TRIAL_MODAL:
      return {
        ...state,
        showInstagramFirstCommentStartTrialModal: true,
      };

    default:
      return state;
  }
};

export const actions = {
  showUpgradeModal: ({ source }) => ({
    type: actionTypes.SHOW_UPGRADE_MODAL,
    source,
  }),
  hideUpgradeModal: () => ({
    type: actionTypes.HIDE_UPGRADE_MODAL,
  }),
  showB4BTrialExpiredModal: ({ source }) => ({
    type: actionTypes.SHOW_UPGRADE_B4B_MODAL,
    source,
  }),
  hideUpgradeB4BModal: () => ({
    type: actionTypes.HIDE_UPGRADE_B4B_MODAL,
  }),
  showInstagramFirstCommentModal: ({ ids }) => ({
    type: actionTypes.SHOW_INSTAGRAM_FIRST_COMMENT_MODAL,
    ids,
  }),
  showInstagramNewFirstCommentUserModal: () => ({
    type: actionTypes.SHOW_INSTAGRAM_NEW_FIRST_COMMENT_USER_MODAL,
  }),
  showWelcomeModal: () => ({
    type: actionTypes.SHOW_WELCOME_MODAL,
  }),
  hideWelcomeModal: () => ({
    type: actionTypes.HIDE_WELCOME_MODAL,
  }),
  showWelcomePaidModal: () => ({
    type: actionTypes.SHOW_WELCOME_PAID_MODAL,
  }),
  hideWelcomePaidModal: () => ({
    type: actionTypes.HIDE_WELCOME_PAID_MODAL,
  }),
  hideInstagramFirstCommentModal: () => ({
    type: actionTypes.HIDE_INSTAGRAM_FIRST_COMMENT_MODAL,
  }),
  hideInstagramNewFirstCommentUserModal: () => ({
    type: actionTypes.HIDE_INSTAGRAM_NEW_FIRST_COMMENT_USER_MODAL,
  }),
  hideInstagramFirstCommentStartTrialModal: () => ({
    type: actionTypes.HIDE_INSTAGRAM_FIRST_COMMENT_START_TRIAL_MODAL,
  }),
  showInstagramFirstCommentStartTrialModal: () => ({
    type: actionTypes.SHOW_INSTAGRAM_FIRST_COMMENT_START_TRIAL_MODAL,
  }),
  showProfilesDisconnectedModal: () => ({
    type: actionTypes.SHOW_PROFILES_DISCONNECTED_MODAL,
  }),
  hideProfilesDisconnectedModal: () => ({
    type: actionTypes.HIDE_PROFILES_DISCONNECTED_MODAL,
  }),
  showStealProfileModal: ({ stealProfileUsername }) => ({
    type: actionTypes.SHOW_STEAL_PROFILE_MODAL,
    stealProfileUsername,
  }),
  hideStealProfileModal: () => ({
    type: actionTypes.HIDE_STEAL_PROFILE_MODAL,
  }),
  showInstagramDirectPostingModal: ({ profileId }) => ({
    type: actionTypes.SHOW_IG_DIRECT_POSTING_MODAL,
    profileId,
  }),
  hideInstagramDirectPostingModal: () => ({
    type: actionTypes.HIDE_IG_DIRECT_POSTING_MODAL,
  }),
  showWelcomeB4BTrialModal: () => ({
    type: actionTypes.SHOW_WELCOME_B4B_TRIAL_MODAL,
  }),
  hideWelcomeB4BTrialModal: () => ({
    type: actionTypes.HIDE_WELCOME_B4B_TRIAL_MODAL,
  }),
};
