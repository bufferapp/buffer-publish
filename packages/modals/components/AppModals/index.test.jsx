import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { Provider } from 'react-redux';

import AppModalsConnected from '../../index';
import AppModals from './index';
import { initialState } from '../../reducer';

const fakeStore = state => ({
  default: () => {},
  subscribe: () => {},
  dispatch: () => {},
  getState: () => ({ ...state }),
});

describe('Modals', () => {
  it('renders nothing when no modals are visible', () => {
    const visibleModals = { ...initialState, showUpgradeModal: false };
    const renderer = new ShallowRenderer();
    const rendered = renderer.render(
      <AppModals {...visibleModals} />,
    );
    expect(rendered).toMatchSnapshot();
  });
  it('renders upgrade modal', () => {
    const visibleModals = { ...initialState, showUpgradeModal: true };
    const renderer = new ShallowRenderer();
    const rendered = renderer.render(
      <AppModals {...visibleModals} />,
    );
    expect(rendered).toMatchSnapshot();
  });
  it('renders a connected upgrade modal', () => {
    const visibleModals = { ...initialState, showUpgradeModal: true };
    const renderer = new ShallowRenderer();
    const rendered = renderer.render(
      <Provider store={fakeStore({ modals: visibleModals })}>
        <AppModalsConnected />
      </Provider>,
    );
    expect(rendered).toMatchSnapshot();
  });
  it('renders welcome modal', () => {
    const visibleModals = { ...initialState, showWelcomeModal: true };
    const renderer = new ShallowRenderer();
    const rendered = renderer.render(
      <AppModals {...visibleModals} />,
    );
    expect(rendered).toMatchSnapshot();
  });
  it('renders a connected welcome modal', () => {
    const visibleModals = { ...initialState, showWelcomeModal: true };
    const renderer = new ShallowRenderer();
    const rendered = renderer.render(
      <Provider store={fakeStore({ modals: visibleModals })}>
        <AppModalsConnected />
      </Provider>,
    );
    expect(rendered).toMatchSnapshot();
  });
  it('renders disconnected profiles modal', () => {
    const visibleModals = { ...initialState, showProfilesDisconnectedModal: true };
    const renderer = new ShallowRenderer();
    const rendered = renderer.render(
      <AppModals {...visibleModals} />,
    );
    expect(rendered).toMatchSnapshot();
  });
  it('renders a connected disconnected profiles modal', () => {
    const visibleModals = { ...initialState, showProfilesDisconnectedModal: true };
    const renderer = new ShallowRenderer();
    const rendered = renderer.render(
      <Provider store={fakeStore({ modals: visibleModals })}>
        <AppModalsConnected />
      </Provider>,
    );
    expect(rendered).toMatchSnapshot();
  });
  it('renders welcome b4b trial modal', () => {
    const visibleModals = { ...initialState, showWelcomeB4BTrialModal: true };
    const renderer = new ShallowRenderer();
    const rendered = renderer.render(
      <AppModals {...visibleModals} />,
    );
    expect(rendered).toMatchSnapshot();
  });
});
