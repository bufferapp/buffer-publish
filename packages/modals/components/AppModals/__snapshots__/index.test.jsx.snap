// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modals renders a connected disconnected profiles modal 1`] = `<Connect(AppModals) />`;

exports[`Modals renders a connected upgrade modal 1`] = `<Connect(AppModals) />`;

exports[`Modals renders a connected welcome modal 1`] = `<Connect(AppModals) />`;

exports[`Modals renders disconnected profiles modal 1`] = `
<React.Fragment>
  <Connect(ProfilesDisconnectedModal) />
</React.Fragment>
`;

exports[`Modals renders nothing when no modals are visible 1`] = `<React.Fragment />`;

exports[`Modals renders upgrade modal 1`] = `
<React.Fragment>
  <Connect(UpgradeModal) />
</React.Fragment>
`;

exports[`Modals renders welcome modal 1`] = `
<React.Fragment>
  <Connect(WelcomeModal) />
</React.Fragment>
`;
